sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -983,28 +983,45 @@ export class ServiceBusMessageImpl implements ReceivedMessageWithLock {
   /**
    * See ReceivedMessageWithLock.deadLetter().
    */
-  async deadLetter(options?: DeadLetterOptions): Promise<void> {
+  async deadLetter(propertiesToModify?: DeadLetterOptions & { [key: string]: any }): Promise<void> {
     const error: AmqpError = {
       condition: Constants.deadLetterName
     };
-    if (options) {
+
+    if (
+      propertiesToModify &&
+      (propertiesToModify.deadLetterReason != null ||
+        propertiesToModify.deadLetterErrorDescription != null)
+    ) {
       error.info = {
-        DeadLetterReason: options.deadletterReason,
-        DeadLetterErrorDescription: options.deadLetterErrorDescription
+        DeadLetterReason: propertiesToModify.deadLetterReason,
+        DeadLetterErrorDescription: propertiesToModify.deadLetterErrorDescription
       };
     }
     log.message(
       "[%s] Deadlettering the message with id '%s'.",
       this._context.namespace.connectionId,
       this.messageId
     );
+
+    const actualPropertiesToModify: Partial<DeadLetterOptions> = {
+      ...propertiesToModify
+    };
+
+    // these two fields are handled specially and don't need to be in here.
+    delete actualPropertiesToModify.deadLetterErrorDescription;
+    delete actualPropertiesToModify.deadLetterReason;
+
+    // this handles deferred messages that have been received via the management link
+    // and thus need to be settled there.
```

Comment:
The other settlement methods also need these comments for deferred-messages-settlement, can you update either in this PR itself or in a new PR?